src/BlazorWebView/src/Maui/Windows/BlazorWebViewHandler.Windows.cs

```diff
@@ -30,13 +31,24 @@ protected override void DisconnectHandler(WebView2Control platformView)
 		{
 			if (_webviewManager != null)
 			{
-				// Dispose this component's contents and block on completion so that user-written disposal logic and
-				// Blazor disposal logic will complete.
-				_webviewManager?
+				// Start the disposal...
+				var disposalTask = _webviewManager?
 					.DisposeAsync()
-					.AsTask()
-					.GetAwaiter()
-					.GetResult();
+					.AsTask()!;
+
+				if (IsBlockingDisposalEnabled)
+				{
+					// If the app is configured to block on dispose via an AppContext switch,
+					// we'll synchronously wait for the disposal to complete. This can cause a deadlock.
+					disposalTask
+						.GetAwaiter()
+						.GetResult();
+				}
+				else
+				{
+					// Otherwise, by default, we'll fire-and-forget the disposal task.
+					disposalTask.FireAndForget();
```

Comment:
Unlike with the other `BlazorWebViewHandler` implementations, there wasn't already an `ILogger` available to pass to the `FireAndForget` call here. If others think it's worth it or necessary to obtain one from the service provider, I won't object to adding it.

Reply:
I think this code is fine.